productivity-suite/piercing-library/src/piercing-fragment-outlet.ts

```diff
@@ -82,6 +82,9 @@ export class PiercingFragmentOutlet extends HTMLElement {
   private pierceFragmentIntoDOM(fragmentHost: PiercingFragmentHost) {
     this.innerHTML == "";
     fragmentHost.pierceInto(this);
+    const fragmentsToPierce =
+      getBus().latestValue<number>("fragmentsToPierce")!;
+    getBus().dispatch("fragmentsToPierce", fragmentsToPierce - 1);
```

Comment:
Why not do this inside the `pierceInto()` method, since there is already access to the message bus there.

Reply:
ah yeah.... good point 😓